server/src/controllers/slips.js

```diff
@@ -0,0 +1,52 @@
+import { Slip } from "../models/slip.js";
+import { Driver } from "../models/driver.js";
+
+class SlipsController {
+  getSlips = async (req, res) => {
+    let conditions = {}
+    if (req.query.driver_id != null) {
+      conditions["driver_id"] = req.query.driver_id
+    }
+    if (req.query.before != null) {
+      conditions["timestamp"] = {$lte: new Date(Number(req.query.before))}
+    }
+    if (req.query.after != null) {
+      conditions["timestamp"] = {$gte: new Date(Number(req.query.after))}
+    }
+
+    Slip.find(conditions)
+      .then(async slips => {
+        if (req.query.company_id != null) {
+          const filters = await Promise.all(slips.map(async (s) => {
+            const d = await Driver.findOne({id: s.driver_id})
+            return (d && d.company_id === req.query.company_id)
+          }));
+          // eslint-disable-next-line
+          return res.status(200).json(slips.filter((s, idx, arr) => {return filters[idx]}))
+        } else {
+          return res.status(200).json(slips)
+        }
+      })
+      .catch(err => {
+        console.log("getSlips: " + err)
+        return res.status(500).json(err)
+      });
+  };
+    
+  getSlipById = async (req, res) => {
```

Comment:
Could you please tell me where the endpoint/ controller to fetch slips based on a date range is? Also, to fetch drivers from drivers table using id.

Reply:
the endpoints and corresponding behaviors are listed in the PR description above
backend is not currently deployed yet, so the only way to actually hit it is to run the server locally